src/jdk.compiler/share/classes/com/sun/tools/javac/code/LintSuppression.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.tools.javac.code;
+
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.sun.tools.javac.code.Lint.LintCategory;
+import com.sun.tools.javac.tree.JCTree.*;
+import com.sun.tools.javac.util.Assert;
+import com.sun.tools.javac.util.Context;
+import com.sun.tools.javac.util.Names;
+
+import static com.sun.tools.javac.code.Lint.LintCategory.DEPRECATION;
+
+/**
+ * Utility class for calculating Lint category suppressions from annotations on a declaration.
+ *
+ *  <p><b>This is NOT part of any supported API.
+ *  If you write code that depends on this, you do so at your own risk.
+ *  This code and its internal interfaces are subject to change or
+ *  deletion without notice.</b>
+ */
+public class LintSuppression {
```

Comment:
Question: why is this a separate class? E.g. as opposed to just add the methods in this class to the main `Lint` class? After all, it seems that `LintSuppression` has no meaningful state so to speak - it seems mostly a collection of "helper methods". I'd use more encapsulation, as these methods should proabably never used outside `Lint` itself?

Reply:
> Question: why is this a separate class?

Simply because it foreshadows [the patch for JDK-8344159](https://github.com/archiecobbs/jdk/blob/suppression/src/jdk.compiler/share/classes/com/sun/tools/javac/code/LintSuppression.java), in which the file becomes a proper singleton responsible for tracking suppressions. Admittedly that's not really a good reason, so I'll fold these methods into `Lint.java` for this patch. Should be fixed in 7cb7aa0cd01.